integration_test/tests/mongosqltranslate_tests.rs

```diff
@@ -0,0 +1,216 @@
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, stmt as Handle)
+            );
+
+            get_column_attributes(stmt as Handle, 5);
+
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLExecute(stmt as HStmt),
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, stmt as Handle)
+            );
+
+            fetch_and_get_data(
+                stmt as Handle,
+                Some(3),
+                vec![SqlReturn::SUCCESS; 5],
+                vec![CDataType::SQL_C_WCHAR; 5],
+            );
+
+            disconnect_and_close_handles(dbc, stmt);
+        }
+        let _ = unsafe { Box::from_raw(env_handle) };
+    }
+
+    #[test]
+    fn test_sql_execute_direct_with_library_loaded_and_valid_query_and_valid_schemas_created() {
+        let (env_handle, dbc, stmt) = default_setup_connect_and_alloc_stmt(
+            AttrOdbcVersion::SQL_OV_ODBC3,
+            Some(crate::common::generate_srv_style_connection_string(Some(
+                "sample_airbnb".to_string(),
+            ))),
+        );
+
+        unsafe {
+            let mut query: Vec<WideChar> = cstr::to_widechar_vec("SELECT property_type, room_type, bed_type, minimum_nights, maximum_nights FROM listingsAndReviews LIMIT 3");
+            query.push(0);
+
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLExecDirectW(stmt as HStmt, query.as_ptr(), SQL_NTS),
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, stmt as Handle)
+            );
+
+            get_column_attributes(stmt as Handle, 5);
+
+            fetch_and_get_data(
+                stmt as Handle,
+                Some(3),
+                vec![SqlReturn::SUCCESS; 5],
+                vec![CDataType::SQL_C_WCHAR; 5],
+            );
+
+            disconnect_and_close_handles(dbc, stmt);
+        }
+        let _ = unsafe { Box::from_raw(env_handle) };
+    }
+
+    #[test]
+    fn test_enterprise_mode_with_library_loaded_and_invalid_query_and_valid_schemas_created() {
+        let (env_handle, dbc, stmt) = default_setup_connect_and_alloc_stmt(
+            AttrOdbcVersion::SQL_OV_ODBC3,
+            Some(crate::common::generate_srv_style_connection_string(Some(
+                "sample_airbnb".to_string(),
+            ))),
+        );
+
+        unsafe {
+            let mut query: Vec<WideChar> =
+                cstr::to_widechar_vec("select * from non_existent_collection");
+            query.push(0);
+
+            assert_eq!(
+                SqlReturn::ERROR,
+                SQLPrepareW(stmt as HStmt, query.as_ptr(), SQL_NTS),
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, stmt as Handle)
+            );
+
+            let error_message = get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, stmt as Handle);
+            assert!(
```

Comment:
It looks like this is more for testing the behaviour for collections with no schema rather than an invalid SQL query which would failed parsing or algebrizing.
Also, it should to fail later if we send empty schemas for collections with no schemas found, no?.

Reply:
I changed the SQL query to `SELECT bed_type.type FROM listingsAndReviews`. 

`bed_type` is a String, so it fails with this error: 
`[MongoDB][Core] The libmongosqltranslate command 'translate' failed. Error message: algebrize error: Error 1002: Incorrect argument type for 'FieldAccess'. Required: object type. Found: string.`.

I think it's fine now?

Reply:
Yes, if it is what you get that looks good. I don't see the change on the PR yet though ;)